API McpRequestHandler

sdelamo opened the pull request:
This pull-request adds a simple `McpRequestHandler` API to handle MCP request is a composite way. 

This pull-request moves the logic to the `micronaut-mcp` module which does not depend on HTTP to support other transports.

sdelamo committed:
API McpRequestHandler

foo

sdelamo committed:
Remove this unused "MCP_METHOD_PING" private field.

graemerocher commented:
A question with the PRs so far. Why aren't we implemented the official MCP Java SDK? https://github.com/modelcontextprotocol/java-sdk/

sdelamo commented:
> A question with the PRs so far. Why aren't we implemented the official MCP Java SDK? https://github.com/modelcontextprotocol/java-sdk/

[I moved this to a discussion](https://github.com/micronaut-projects/micronaut-mcp/discussions/9).
